ttnn/cpp/ttnn/decorators.hpp

```diff
@@ -82,7 +82,22 @@ inline Tensors create_async_output_tensors(const Tensors& inputs, const Optional
     };
 
     if constexpr (custom_create_async_outputs) {
-        return concrete_operation_t::create_async_output_tensors(inputs, optional_inputs);
+        auto output = concrete_operation_t::create_async_output_tensors(inputs, optional_inputs);
```

Comment:
Remove the changes from line 85 to line 100. If you are using custom `create_async_output_tensors`, you should handle everything there. Don't modify generic code please use `execute_on_main_thread` and remove all changes from `decorators.hpp`

Reply:
I have reverted the changes in the decorators.hpp, kindly review it once